src/cfnlint/helpers.py

```diff
@@ -165,10 +166,52 @@
     'AWS::Redshift::Cluster'
 ]
 
+CACHING_DIR = os.path.join(os.path.dirname(__file__), 'cache/')
```

Comment:
This can easily be changed to a different directory or moved into a config option if desired. I'm also unsure of the consequences of using `__file__` rather than `cfnlint.__file__` as is used in some places, but there doesn't seem to be a consistent approach being used, and there were circular dependency issues with the latter option.

Reply:
Wondering if it should be under the data folder?  Would we want to commit that file as we do the `update-specs` so the most recent version is included in the package?

Reply:
I'm happy for it to live under the data folder; the primary reason for using a separate cache directory was just to signify the temporary, non-essential nature of the files. It definitely could be commited and used like a lock file, and would save an initial fetch of every spec file when it's first run.